pkg/storegateway/chunkscache/cache.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkscache
+
+import (
+	"context"
+
+	gotkit_log "github.com/go-kit/log"
+	"github.com/oklog/ulid"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+type Range struct {
+	BlockID   ulid.ULID
+	Start     chunks.ChunkRef
+	NumChunks int
+}
+
+type ChunksCache interface {
+	FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte)
+	StoreChunks(ctx context.Context, userID string, r Range, v []byte)
+}
+
+type TracingCache struct {
+	C ChunksCache
+}
+
+func (c TracingCache) FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte) {
+	l := spanlogger.FromContext(ctx, gotkit_log.NewNopLogger())
```

Comment:
I would recommend to not create a new logger each time. Pass the actual logger to `NewTracingCache()`. We have it in the caller.

Reply:
i thought the spanlogger is different for each context. But we only create the TracinCache once on startup. We also use a spanlogger heavily for the [index cache for example](https://github.com/grafana/mimir/blob/e7807e00a449de34e1535b7efca954f1b9b2524d/pkg/storegateway/bucket.go#L747-L758)

Reply:
My comment is about `gotkit_log.NewNopLogger()`.